internal/provider/resource_vault_cluster_perf_replication_test.go

```diff
@@ -0,0 +1,330 @@
+			},
+			{
+				// secondary cluster created successfully (same hvn)
+				Config: testConfig(setTestAccPerformanceReplicationE2E(`
+				resource "hcp_vault_cluster" "c1" {
+					cluster_id      = "test-primary"
+					hvn_id          = hcp_hvn.hvn1.hvn_id
+					tier            = "plus_small"
+					public_endpoint = true
+				}
+				resource "hcp_vault_cluster" "c2" {
+					cluster_id   = "test-secondary"
+					hvn_id       = hcp_hvn.hvn1.hvn_id
+					tier         = lower(hcp_vault_cluster.c1.tier)
+					primary_link = hcp_vault_cluster.c1.self_link
+					paths_filter = ["path/a", "path/b"]
+				}
+				`)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVaultClusterExists(primaryVaultResourceName),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "cluster_id", "test-secondary"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "hvn_id", "test-perf-hvn-1"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "tier", "PLUS_SMALL"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "cloud_provider", "aws"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "region", "us-west-2"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "public_endpoint", "false"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "paths_filter.0", "path/a"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "paths_filter.1", "path/b"),
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "namespace", "admin"),
+					resource.TestCheckResourceAttrSet(secondaryVaultResourceName, "vault_version"),
+					resource.TestCheckResourceAttrSet(secondaryVaultResourceName, "organization_id"),
+					resource.TestCheckResourceAttrSet(secondaryVaultResourceName, "project_id"),
+					resource.TestCheckNoResourceAttr(secondaryVaultResourceName, "vault_public_endpoint_url"),
+					resource.TestCheckResourceAttrSet(secondaryVaultResourceName, "self_link"),
+					resource.TestCheckResourceAttrSet(secondaryVaultResourceName, "vault_private_endpoint_url"),
+					testAccCheckFullURL(secondaryVaultResourceName, "vault_private_endpoint_url", ""),
+					resource.TestCheckResourceAttrSet(secondaryVaultResourceName, "created_at"),
+				),
+			},
+			{
+				// update paths filter
+				Config: testConfig(setTestAccPerformanceReplicationE2E(`
+				resource "hcp_vault_cluster" "c1" {
+					cluster_id      = "test-primary"
+					hvn_id          = hcp_hvn.hvn1.hvn_id
+					tier            = "plus_small"
+					public_endpoint = true
+				}
+				resource "hcp_vault_cluster" "c2" {
+					cluster_id   = "test-secondary"
+					hvn_id       = hcp_hvn.hvn1.hvn_id
+					tier         = hcp_vault_cluster.c1.tier
+					primary_link = hcp_vault_cluster.c1.self_link
+					paths_filter = ["path/a", "path/c"]
+				}
+				`)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(secondaryVaultResourceName, "paths_filter.0", "path/a"),
```

Comment:
Test design question: I've been worried about the amount of time this test takes-- typically over an hour. I've often run into timeouts when testing the entire acceptance suite locally.

I appreciate that you've added the Vault cluster tests in parallel. :) But looking at this performance replication test, I wonder if we might save some test time by simplifying the steps to cover a single happy path, like a single field update. The other edge cases should certainly be covered, but with the provider being responsible for testing so many different HCP resources on every release, this repo might not be the best place for them.

Reply:
Our team is working on ways we might cache acceptance test results, so that the entire suite need not be run for unrelated changes. That may help alleviate this issue as well.

Reply:
@bcmdarroch Lets sync tomorrow on this. I hear your concern, but I know we want to test all of these paths. Now can choose to do it for one provider, but our AWS and Azure code paths are not same hence we need to test both. Maybe we can find common ground to change some of our test suites to make the experience better. In PRDE it was really hard to test for me as well. not sure if you tested in any other env and had this experience.

Reply:
After chatting with @bcmdarroch and @himran92, we decided to keep moving forward. @bcmdarroch and team will look into running per replication test optionally vs every time so that pipeline is not blocked.